widget/embedded/src/components/HeaderButtons/RefreshButton.tsx

```diff
@@ -0,0 +1,60 @@
+import { RefreshProgressIcon } from '@rango-dev/ui';
+import React, { useEffect, useState } from 'react';
+
+import { HeaderButton, ProgressIcon } from './HeaderButtons.styles';
+
+const REFRESH_INTERVAL = 1000;
+const MAX_ELAPSED_TIME = 60;
+const MAX_PERCENTAGE = 100;
+
+function RefreshButton({ onClick }: { onClick: (() => void) | undefined }) {
+  const [elapsedTime, setElapsedTime] = useState(0);
+  const [isRefetched, setIsRefetched] = useState(false);
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timeout | null = null;
```

Comment:
I think this type is more appropriate for browser: 
let interval: number;
interval = window.setInterval

Reply:
fixed.